### PR TITLE
Windows: install dotnet and latest wix

### PR DESCRIPTION
**Pull request by lsm5**
wix3 is EOL and choco doesn't support installing wix > 3.14.
    
So, this commit installs the `dotnet` runtime and uses dotnet to install
the latest wix in the windows image.
    
Resolves: RUN-2055

### CODE REVIEW

File: win_images/win_packaging.ps1
Comment (cevich) [inline win_images/win_packaging.ps1:34]: Assuming what you added above works, my preference would be to just delete all this commented out stuff.  Otherwise anybody else finding this is going to wonder "why" - which isn't commented anywhere.  It's okay to rely on git to remember these commands if we ever need them again.

**Review by cevich — APPROVED** [review]
LGTM - just one minor comment.

**Comment by cevich** [comment]
Assuming you want to build AWS arm Fedora.

File: win_images/win_packaging.ps1
Comment (lsm5) [reply win_images/win_packaging.ps1:34]: Ack, I'll remove those after testing the built image in the podman windows installer PR. Thanks a lot

**Comment by lsm5** [comment]
Win-images build successful, now to wait to test the new image id in the podman PR.

File: win_images/win_packaging.ps1
Comment (edsantiago) [inline win_images/win_packaging.ps1:17]: This is an "add one to X" comment. If you need to repush, could you please use comments to explain the why, not the what? Even something like the previous comment ("needed for wixtoolset") would be better. Remember that this code may need to be maintained some day by non-Windows-experts. They might be reassured by a short sentence or two describing what these things are and why they're needed.

**Comment by edsantiago** [comment]
Comparing against the versions in podman right now:

- 20240509t162134z-f40f39d13
- 20240506t132946z-f40f39d13 &#x21D1; (built in #349)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*kernel*       | 6.7.12-1        | 6.8.8-200       | 6.8.9-300       | 6.8.8-300       | 6.9.0-0.rc7.20240507gitdccb07f2914c.59 |
|               |                |                |                |                | 6.9.0-0.rc6.20240503gitf03359bca01b.56&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.9-1        | 1.22.3-1        | 1.22.2-1        | 1.22.2-3        |
|               |                |                | 1.22.2-1&nbsp;&#x21D1; |                |                |

**Comment by lsm5** [comment]
> Comparing against the versions in podman right now:
> 
> * 20240509t162134z-f40f39d13
> * 20240506t132946z-f40f39d13 ⇑ (built in [Fedora40 #349](https://github.com/containers/automation_images/pull/349))
> 
> _debian_	_prior-fedora_	_fedora_	_fedora-aws_	_rawhide_
> _kernel_	6.7.12-1	6.8.8-200	6.8.9-300	6.8.8-300	6.9.0-0.rc7.20240507gitdccb07f2914c.59
> 6.9.0-0.rc6.20240503gitf03359bca01b.56 ⇑
> _golang_	2:1.22~3	1.21.9-1	1.22.3-1	1.22.2-1	1.22.2-3
> 1.22.2-1 ⇑

Are the new rawhide kernel and fedora golang versions going to be trouble?

File: win_images/win_packaging.ps1
Comment (lsm5) [reply win_images/win_packaging.ps1:17]: will do

**Comment by edsantiago** [comment]
Very very unlikely.

File: win_images/win_packaging.ps1
Comment (lsm5) [reply win_images/win_packaging.ps1]: @edsantiago addressed this in the latest update. PTAL.

**Comment by edsantiago** [comment]
Better, thank you!

**Comment by lsm5** [comment]
hmm, looks like I need to add an additional package https://www.nuget.org/packages/WixToolset.Heat in the windows image too. ughhhh

**Comment by edsantiago** [comment]
(No package diffs against previous build)

File: win_images/win_packaging.ps1
Comment (edsantiago) [inline win_images/win_packaging.ps1:18]: If you need to resubmit (for the `heat` thing), might as well `s/anythong/anything/`

**Comment by cevich** [comment]
FYI: Since Ed and I will both be gone Friday, and myself all of next week I said it's okay for Lokesh to merge this PR assuming the run through podman CI passes.

**Comment by edsantiago** [comment]
- 20240510t174901z-f40f39d13
- 20240509t185622z-f40f39d13 &#x21D1; (built in #354)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*kernel*       | 6.7.12-1        | 6.8.9-200       | 6.8.9-300       | 6.8.9-300       | 6.9.0-0.rc7.20240509git45db3ab70092.61 |
|               |                | 6.8.8-200&nbsp;&#x21D1; |                | 6.8.8-300&nbsp;&#x21D1; | 6.9.0-0.rc7.20240507gitdccb07f2914c.59&nbsp;&#x21D1; |
|*golang*       | 2:1.22~3        | 1.21.9-1        | 1.22.3-1        | 1.22.2-1        | 1.22.3-1        |
|               |                |                |                |                | 1.22.2-3&nbsp;&#x21D1; |
|*systemd*      | 255.5-1         | 254.10-1        | 255.6-1         | 255.4-1         | 255.5-1         |
|               |                |                | 255.4-1&nbsp;&#x21D1; |                |                |

**Comment by cevich** [comment]
@edsantiago @lsm5 FYI: when these images are ready, I'd like to either tag them or manually bring them over to https://github.com/containers/common/pull/2024 since they will fix a failing test over there.

**Comment by edsantiago** [comment]
SGTM. Here are the diffs against what's currently in podman. I _think_ the rawhide systemd (4) is a good version.

- 20240612t140514z-f40f39d13
- 20240529t141726z-f40f39d13 &#x21D1; (built in #338)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13.3            | 39-1.5          | Generic         | ?               | 41-0            |
|               | 13.2&nbsp;&#x21D1; |                |                |                |                |
|*kernel*       | 6.8.12-1        | 6.8.11-200      | 6.8.5-301       | 6.8.5-301       | 6.8.5-301       |
|               | 6.8.11-1&nbsp;&#x21D1; | 6.8.10-200&nbsp;&#x21D1; |                |                | 6.9.0-64&nbsp;&#x21D1; |
|*grub2-common* | 2.12-2          | 2.06-120        | 2.06-123        | 2.06-123        | 2.06-123        |
|               |                |                | 2.06-121&nbsp;&#x21D1; | 2.06-121&nbsp;&#x21D1; | 2.06-121&nbsp;&#x21D1; |
|*aardvark-dns* | 1.4.0-5.1       | 1.10.0-1        | 1.11.0-1        | 1.10.0-1        | 1.11.0-1        |
|               |                |                | 1.10.0-1&nbsp;&#x21D1; |                | 1.10.0-1&nbsp;&#x21D1; |
|*netavark*     | 1.4.0-4.1       | 1.10.3-1        | 1.11.0-1        | 1.10.3-3        | 1.10.3-3        |
|               |                |                | 1.10.3-3&nbsp;&#x21D1; |                |                |
|*buildah*      | 1.33.7+ds1-1    | 1.35.4-1        | 1.36.0-1        | 1.35.4-1        | 1.36.0-1        |
|               |                |                |                |                | 1.35.4-1&nbsp;&#x21D1; |
|*conmon*       | 2.1.10+ds1-1+b1 | 2.1.12-1        | 2.1.12-1        | 2.1.10-1        | 2.1.10-1        |
|               |                | 2.1.10-1&nbsp;&#x21D1; |                |                |                |
|*container-selinux*| ?               | 2.231.0-1       | 2.232.1-1       | 2.231.0-1       | 2.231.0-6       |
|               |                |                | 2.231.0-1&nbsp;&#x21D1; |                |                |
|*containers-common*| ?               | 1-99            | 0.59.1-2        | 0.59.1-1        | 0.59.1-1        |
|               |                |                | 0.59.0-1&nbsp;&#x21D1; | 0.58.0-2&nbsp;&#x21D1; | 0.59.0-1&nbsp;&#x21D1; |
|*docker-ce*    | 5:26.1.4-1~debian.12~bookworm | ?               | ?               | ?               | ?               |
|               | 5:26.1.3-1~debian.12~bookworm&nbsp;&#x21D1; |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.10-1       | 1.22.4-1        | 1.22.4-1        | 1.22.3-1        |
|               |                |                | 1.22.3-1&nbsp;&#x21D1; | 1.22.3-1&nbsp;&#x21D1; |                |
|*nmap-ncat*    | 7.94+git20230807.3be01efb1+dfsg-4 | 7.95-1          | 7.95-1          | 7.95-1          | 7.95-1          |
|               | 7.94+git20230807.3be01efb1+dfsg-3+b1&nbsp;&#x21D1; |                |                |                |                |
|*passt*        | 2024-06-07      | 2024-05-10      | 2024-06-07      | 2024-05-10      | 2024-06-07      |
|               | 2024-04-26&nbsp;&#x21D1; |                | 2024-05-10&nbsp;&#x21D1; |                | 2024-05-23&nbsp;&#x21D1; |
|*podman*       | 4.9.4+ds1-1     | 4.9.4-1         | 5.1.1-1         | 5.1.0-1         | 5.0.3-1         |
|               |                |                | 5.1.0~rc1-1&nbsp;&#x21D1; | 5.0.3-1&nbsp;&#x21D1; |                |
|*skopeo*       | 1.13.3+ds1-2+b1 | 1.15.1-1        | 1.15.1-1        | 1.15.1-1        | 1.15.1-1        |
|               |                | 1.15.0-1&nbsp;&#x21D1; |                | 1.15.0-1&nbsp;&#x21D1; |                |
|*systemd*      | 256-1           | 254.13-1        | 255.7-1         | 255.7-1         | 256~rc4-2       |
|               | 256~rc3-5&nbsp;&#x21D1; | 254.12-1&nbsp;&#x21D1; |                |                | 256~rc3-1&nbsp;&#x21D1; |

**Comment by edsantiago** [comment]
OOPS! You better not use these images in podman, because of the `ima` bug. Please grab the `setfattr` timebomb from #357. You will also need the `Makefile` fix.

**Comment by lsm5** [comment]
@edsantiago only windows specific. And won't be merged until the wix update PR is finalized.

**Comment by edsantiago** [comment]
@lsm5 what is the status of this? If you rebase on main, you should end up with images that can be tagged for update into all containers repos. (Assuming this solves the Windows problem you're trying to solve)

**Comment by lsm5** [comment]
> @lsm5 what is the status of this? If you rebase on main, you should end up with images that can be tagged for update into all containers repos. (Assuming this solves the Windows problem you're trying to solve)

status is not ready until the windows installer with wix update is ready. But please ignore until further notice.

**Comment by lsm5** [comment]
@cevich @edsantiago https://github.com/containers/podman/pull/22408 has passed CI but still pending review. Remind me, should I wait for that PR to merge first before merging this image update PR?

**Comment by edsantiago** [comment]
I have a medium-high preference for merging an auto-imgs PR before a tool-that-uses-it PR, but as long as both get done shortly within each other I think it's OK

**Comment by lsm5** [comment]
ok, I'll set up a hold on the podman PR and remove it only after we have 1. LGTM there and 2. auto-img pr is merged.

**Comment by edsantiago** [comment]
- 20240626t192556z-f40f39d13
- 20240620t153000z-f40f39d13 &#x21D1; (built in #361)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*kernel*       | 6.8.12-1        | 6.9.5-100       | 6.9.6-200       | 6.9.5-200       | 6.8.5-301       |
|               |                | 6.9.4-100&nbsp;&#x21D1; | 6.9.5-200&nbsp;&#x21D1; | 6.9.4-200&nbsp;&#x21D1; |                |
|*grub2-common* | 2.12-2          | 2.06-121        | 2.06-123        | 2.06-123        | 2.06-123        |
|               |                | 2.06-120&nbsp;&#x21D1; |                |                |                |
|*aardvark-dns* | 1.4.0-6         | 1.11.0-1        | 1.11.0-1        | 1.11.0-1        | 1.11.0-1        |
|               | 1.4.0-5.1&nbsp;&#x21D1; |                |                |                |                |
|*docker-ce*    | 5:27.0.1-1~debian.12~bookworm | ?               | ?               | ?               | ?               |
|               | 5:26.1.4-1~debian.12~bookworm&nbsp;&#x21D1; |                |                |                |                |
|*golang*       | 2:1.22~3        | 1.21.11-1       | 1.22.4-1        | 1.22.4-1        | 1.22.4-4        |
|               |                |                |                |                | 1.22.4-3&nbsp;&#x21D1; |
|*passt*        | 2024-06-24      | 2024-06-07      | 2024-06-24      | 2024-06-07      | 2024-06-24      |
|               | 2024-06-07&nbsp;&#x21D1; | 2024-05-10&nbsp;&#x21D1; | 2024-06-07&nbsp;&#x21D1; |                | 2024-06-07&nbsp;&#x21D1; |
|*systemd*      | 256.1-2         | 254.13-1        | 255.7-1         | 255.7-1         | 256.1-3         |
|               | 256.1-1&nbsp;&#x21D1; |                |                |                |                |

**Comment by cevich** [comment]
SGTM

File: win_images/win_packaging.ps1
Comment (l0rd) [inline win_images/win_packaging.ps1:20]: I would have installed it using `winget` because it automatically adds `dotnet` to the `%PATH%`. Using the script works too, though, and there may be other reasons to choose it that I am not aware of. 

What I have [found ](https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script?WT.mc_id=dotnet-35129-website#purpose)is:

```
The intended use of the scripts is for Continuous Integration (CI) scenarios, where:
- The SDK needs to be installed without user interaction and without admin rights.
- The SDK installation doesn't need to persist across multiple CI runs.

The typical sequence of events:
. CI is triggered.
. CI installs the SDK using one of these scripts.
. CI finishes its work and clears temporary data including the SDK installation.

To set up a development environment or to run apps, use the installers rather than these scripts.
```

In the `build_windows.md`[ we are suggesting to install dotnet using winget](https://github.com/containers/podman/pull/22408/files#diff-7208b93342b250997fed5737592fbeec1afdb6271b12ef7e10a12a9200e783c1R80).

File: win_images/win_packaging.ps1
Comment (l0rd) [inline win_images/win_packaging.ps1:23]: Locally, I didn't had to do that. `dotnet tool install --global wix` worked right after `dotnet` installation. No extra configuration of sources was required. That may be related to the use of winget instead of the pwsh script though.

File: win_images/win_packaging.ps1
Comment (l0rd) [inline win_images/win_packaging.ps1:30]: Also, the two commands above were not required in my case. I could build the msi package without running them.

**Review by l0rd — APPROVED** [review]
LGTM with some non-blocking comments.

File: win_images/win_packaging.ps1
Comment (lsm5) [reply win_images/win_packaging.ps1]: I pushed a separate commit here to use winget. If this works out, even better. Thanks.

Comment (cevich) [reply win_images/win_packaging.ps1]: Drive-by non-expert comment:  Remember this is preparing a VM image to run CI w/o having to install anything during runtime.  So the "typical sequence" isn't completely correct in this context.  This makes me think that having `%PATH%` all ready to go at CI runtime might be a good idea - it will simplify the CI scripts to which more developers are exposed.

(please take this comment with a grain of salt)

**Review by l0rd — APPROVED** [review]

**Comment by lsm5** [comment]
I'm gonna merge this soon after this passes.

**Comment by edsantiago** [comment]
FWIW, this got a bunch of "Context canceled" failures. I got that earlier today and had to repush because something-something images had already been tagged. I think it's a Cirrus issue. All the jobs got retried here and passed.

Diff against Chris's PR from yesterday:

- 20240702t180702z-f40f39d13
- 20240701t155130z-f40f39d13 &#x21D1; (built in #363)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*netavark*     | 1.4.0-4.1       | 1.11.0-1        | 1.11.0-1        | 1.11.0-1        | 1.11.0-1        |
|               |                |                |                |                | 1.10.3-3&nbsp;&#x21D1; |

**Comment by edsantiago** [comment]
Looks like Win Installer is broken: https://github.com/containers/podman/pull/22726

**Pull request merged**
